keps/0034-instance-health.md

```diff
@@ -0,0 +1,41 @@
+---
+kep-number: 34
+short-desc: Provide a way to monitor health of KUDO instance post-deployment
+title: KUDO instance health
+authors:
+  - "@alenkacz"
+owners:
+  - "@alenkacz"
+editor: TBD
+creation-date: 2020-09-24
+last-updated: 2020-09-24
+status: provisional
+---
+
+# KUDO instance health
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+
+## Summary
+
+KUDO helps people implement their operators and it's focus is day 2 operations. Part of day 2 is also monitoring your workload health after deployment. To help with that, KUDO will expose "health" computed as a heuristic based on health and readiness of the underlying resources. In the first iteration, health will be just a simple heuristic computed from Pods, StatefulSets, Deployments, ReplicaSets, DaemonSets and Services (let's call them *health phase 1 resources*).
+
+## Motivation
+
+Added health monitoring for `Instance` CRD will help people answer question "is my operator working at this point in time" without querying all underlying resources. KUDO will expose this heuristic as part of `Status` field for everyone to query. This could be used as a signal for a monitoring tool.
+
+### Goals
+
+Expose health heuristic in `Status` field of `Instance`
+Compute health by evaluating health and readiness of *health phase 1 resources*
+
```

Comment:
is it... or is it not a goal... to determine the defined states for the status?

Reply:
I think it will be a boolean right now, but that's implementation detail. So I actually think it's a goal

Reply:
Yeah, I think it should be a goal - and Probably not a boolean? To make it extendable? I think we should at least mirror the Deployment/Statefulset status of "NOT_READY, DEGRADED, RUNNING" or something similar... 

But this is implementation detail, and shouldn't come in at this stage of the KEP, I agree :)

Reply:
Yeah I am going with boolean for now for some reasons that will be explained in the next part of KEP, let's discuss there